### PR TITLE
Update pi_setup doc

### PR DESCRIPTION
**Pull request by jsilke**
Over the weekend, I took some time to configure a new Pi4 as an access point using `dora`. While I definitely found the existing documentation very helpful, I made a few changes to reflect my experience with the setup that will hopefully be helpful to others. Please let me know if there are any further changes I should make (or revert)!

### CODE REVIEW

File: docs/pi_setup.md [hunk 1]
```diff
@@ -1,35 +1,60 @@
...
+```bash
 sudo apt-get -y install hostapd bridge-utils iptables gettext libdbus-1-dev libidn11-dev libnetfilter-conntrack-dev nettle-dev netfilter-persistent iptables-persistent
 ```
 
-## set up the pi as an access point
+Also, if you have not yet installed Rust on your Pi, this can be achieved rather painlessly:
 
-I followed the guide [here](https://www.raspberrypi.com/documentation/computers/configuration.html#setting-up-a-routed-wireless-access-point)
+```bash
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
```
Comment (leshow) [inline docs/pi_setup.md:23]: Let's link to the  `rustup.rs` site for this, I think

File: docs/pi_setup.md [hunk 2]
```diff
@@ -1,35 +1,60 @@
...
+```
+
+Furthermore, you may wish to also install `docker` if you have not yet done so:
```
Comment (leshow) [inline docs/pi_setup.md:26]: Docker is necessary only for cross compiling with `cross` so maybe we should put this in the main README under that section

File: docs/pi_setup.md [hunk 3]
```diff
@@ -40,74 +65,95 @@ rsn_pairwise=CCMP
...
-(use --help to see `dora` arguments)
+You may wish to save this minimal config to `pi.yaml` to try it out, or see [example.yaml](../example.yaml) for the full set of options. You can also use `dora --help` to see arguments.
 
 Run dora:
 
+After you have saved the above minimal config to `pi.yaml`, you should be able to run the following (assuming you compiled a release binary with `armv7-unknown-linux-gnueabihf` as a target earlier, if not please alter the path to your `dora` bin appropriately):
+
 ```
-sudo DORA_LOG="debug" ./dora -c example.yaml -d em.db
+sudo DORA_LOG="debug" target/armv7-unknown-linux-gnueabihf/release/dora -c pi.yaml -d em.db
```
Comment (leshow) [inline docs/pi_setup.md:156]: If you followed this guide to the letter, you'd have cross-compiled a binary and copied to the pi, so this path won't exist.

File: docs/pi_setup.md (hunk 2 above)
Comment (jsilke) [reply docs/pi_setup.md:26]: I agree that this is probably the better way to go 👍

File: docs/pi_setup.md (hunk 1 above)
Comment (jsilke) [reply docs/pi_setup.md:23]: I agree that linking to `rustup.rs`  is a better idea.

File: README.md [hunk 4]
```diff
@@ -81,14 +81,12 @@ Notice that `.cargo/config.toml` has an entry for replacing the linker when cros
 linker = "arm-linux-gnueabihf-gcc"
 ```
 
-This means `arm-linux gnueabihf-gcc` must be available on the system and will be used as the linker. Once you have it installed, you can use:
+This means `arm-linux gnueabihf-gcc` must be available on the system and will be used as the linker. Once you have it installed, you can produce an ARMv7 binary using:
```
Comment (leshow) [inline README.md:84]: there's a missing '-' in `arm-linux-gnueabihf-gcc`.

File: docs/pi_setup.md [hunk 5]
```diff
@@ -40,74 +52,95 @@ rsn_pairwise=CCMP
...
+
+```bash
+sudo iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
+sudo netfilter-persistent save
+```
+
 ## Set up & run dora/hostapd
 
-1. get yourself a dora ARM binary. See the [README](../README.md) the section "Cross Compiling to ARM"
+1. get yourself a dora ARM binary. See the [README](../README.md#cross-compiling-to-arm) the section "Cross Compiling to ARM". Note that if you choose to build the binary directly on the Pi, you may find it easier to [not use cross](../README.md#not-using-cross).
```
[the remark this answers is not in the document]
Comment (jsilke) [reply docs/pi_setup.md:97]: 🤦 of course you are correct here. I don't know why this didn't occur to me to begin with. I now see the silliness of my mistake quite clearly. My apologies, I will fix this.

File: docs/pi_setup.md (hunk 3 above)
Comment (jsilke) [reply docs/pi_setup.md]: I do agree with your point here. This info has been moved to a more relevant section of the README and a note was added about ensuring that `./dora` refers to your binary path.

File: README.md (hunk 4 above)
Comment (jsilke) [reply README.md:84]: Nice catch, I missed that the first time through as well. I will fix this 👍

**Comment by leshow** [comment]
Thank you!

**Comment by leshow** [comment]
Thank you!

**Pull request merged**